jackrabbit-core/src/main/java/org/apache/jackrabbit/core/security/user/UserManagerImpl.java

```diff
@@ -911,6 +933,13 @@ private Authorizable internalGetAuthorizable(String id) throws RepositoryExcepti
         return getAuthorizable(n);
     }
 
+    private <T> T getAuthorizableByType(Authorizable authorizable, Class<T> authorizableClass) throws AuthorizableTypeException {
+        if (authorizable == null || authorizableClass.isInstance(authorizable)) {
+            return authorizableClass.cast(authorizable);
+        }
+        throw new AuthorizableTypeException("Invalid authorizable type '" + authorizable.getClass() + "'");
```

Comment:
potential `NullPointerException`, if `authorizable` is `null`. Wouldn't it be easier to catch the `ClassCastException` ?